review/get_review_ids.py

```diff
@@ -0,0 +1,235 @@
+            vertical_id = parent.block_id
+            vertical_data.add(vertical_id)
+            delete_state_of_review_problem(user, current_course, block_key.block_id)
+
+    if not vertical_data:
+        return []
+
+    vertical_to_show = random.sample(vertical_data, 1)[0]
+    review_course_id = REVIEW_COURSE_MAPPING[str(current_course)]
+    return (TEMPLATE_URL.format(course_id=review_course_id,
+                                type='vertical', xblock_id=vertical_to_show))
+
+
+def get_records(user, current_course):
+    '''
+    Generator that yields each applicable record from the Courseware Student
+    Module. Each record corresponds to a problem the user has loaded
+    in the original course.
+
+    Parameters:
+        user (django.contrib.auth.models.User): User object for the current user
+        current_course (CourseLocator): The course the learner is currently in
+
+    Returns:
+        record.module_state_key (opaque_keys.edx.locator.BlockUsageLocator):
+            The locator for the problem
+        state (dict): The state of the problem
+    '''
+    problem_filter = {'student_id': user.id, 'course_id': current_course, 'module_type': 'problem'}
+    for record in StudentModule.objects.filter(**problem_filter):
+        state = json.loads(record.state)
+        # The key 'selected' shows up if a problem comes from a
+        # library content module. These cause issues so we skip this.
+        # Issue: Library content contains problems but the CSM brings up
+        # the library content and not the problems within
+        if 'selected' not in state:
+            yield record.module_state_key, state
+
+
+def enroll_user_in_review_course(user, current_course):
+    '''
+    If the user is not enrolled in the review version of the course,
+    they are unable to see any of the problems. This ensures they
+    are enrolled so they can see review problems.
+
+    Parameters:
+        user (User): the current user interacting with the review XBlock
+        current_course (CourseLocator): The course the learner is currently in
+    '''
+    enrollment_course_id = ENROLLMENT_COURSE_MAPPING[str(current_course)]
+    enrollment_status = get_enrollment(user.username, enrollment_course_id)
+    if not enrollment_status:
+        add_enrollment(user.username, enrollment_course_id)
+    elif not enrollment_status['is_active']:
+        update_enrollment(user.username, enrollment_course_id, is_active=True)
+
+
+def delete_state_of_review_problem(user, current_course, problem_id):
+    '''
+    Deletes the state of a review problem so it can be used infinitely
+    many times.
+
+    Parameters:
+        user (User): the current user interacting with the review XBlock
+        current_course (CourseLocator): The course the learner is currently in
+        problem_id (str): The problem id whose state should be cleared
+    '''
+    review_course = current_course.replace(course=current_course.course+'_review')
```

Comment:
I'm wondering whether just appending `_review` would be sufficient in the long-run - and whether you would run into conflicts by not having a more unique distinguisher.  But maybe it's Ok.